tests/cxx/explicit_instantiation2.cc

```diff
@@ -0,0 +1,99 @@
+//===------- explicit_instantiation2.cc - test input file for iwyu --------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "explicit_instantiation-template_direct.h"
+#include "explicit_instantiation2-template_helpers.h"
+#include "explicit_instantiation2-template_short_direct.h"
+
+// These tests verify that a dependent explicit instantiation declaration is
+// always required when the main template is also needed, but not otherwise.
+//
+// Positive scenarios:
+// 1a/1b: Implicit instantiation before and after an explicit instantiation
+//        definition (2), as it affects the semantics.
+// 2: Explicit instantiation definition.
+// 3: Full use in a template, provided as an explicit parameter.
+// 4: Fwd-decl use in a template, provided as an explicit parameter.
+// 5: Full use in a template, provided as an default parameter.
+// 7: Full use in a template, provided as a template template parameter.
+// 8: Fwd-decl use in a template, provided as a template template parameter.
+//
+// Negative scenarios, where the dependent template specialization is not
+// required, or it does not provide an explicit instantiation:
+// 6: Fwd-decl use in a template, provided as a default parameter.
+// 9: Implicit instantiation of Template<int>
+//10: Specialization of Template<T>
+
+
+// IWYU: Template is...*explicit_instantiation-template.h
+// IWYU: Template is...*template_short.h.*for explicit instantiation
+Template<short> t1a;
```

Comment:
Ah, now I see you've used naming to refer back to the list... Maybe better to use comments consistently, like `// 2` below?